homeassistant/components/switchbot/switch.py

```diff
@@ -118,25 +103,15 @@ def __init__(
         idx: str | None,
         mac: str,
         name: str,
-        password: str,
+        password: str | None,
         retry_count: int,
     ) -> None:
         """Initialize the Switchbot."""
-        super().__init__(coordinator)
-        self._idx = idx
-        self._last_run_success: bool | None = None
-        self._mac = mac
+        super().__init__(coordinator, idx, mac, name)
+        self._attr_unique_id = self._mac.replace(":", "")
         self._device = self.coordinator.switchbot_api.Switchbot(
             mac=mac, password=password, retry_count=retry_count
         )
```

Comment:
It would be cleaner to create the device in `async_setup_entry` and pass it to `SwitchBotCurtainEntity`

Reply:
Done.